File: CSHARP/Cap08/c08ex04.cs
// c08ex04.cs

using System;
using static System.Console;

class Program
{
    static void Main(string[] args)
    {
        string[] Nomes = new string[5];
        string X;

        int I, J;

        WriteLine("CLASSIFICACAO DE NOMES (DECRESCENTE)\n");

        for (I = 0; I <= 4; ++I)
        {
            Write($"Entre o {I + 1}o. nome: ");
            Nomes[I] = ReadLine();
        }

        for (I = 0; I <= 3; ++I)
            for (J = I + 1; J <= 4; ++J)
                if (Nomes[I].CompareTo(Nomes[J]) > 0)
                {
                    X = Nomes[I];
                    Nomes[I] = Nomes[J];
                    Nomes[J] = X;
                }

        WriteLine();
        for (I = 0; I <= 4; ++I)
            WriteLine($"{I + 1}o. nome: {Nomes[I]}");

        Write("\nTecle <Enter> para encerrar... ");
        ReadLine();
    }
}

File: KOTLIN/Cap08/c08ex04.kt
// c08ex04.kt

fun main() {
    var nomes = Array<String>(5) { "" }
    var x: String

    println("CLASSIFICACAO DE NOMES (DECRESCENTE)\n")

    for (i in 0..4) {
        print("${i + 1}o. nome: ")
        nomes[i] = readLine() ?: ""
    }

    for (i in 0..3) 
        for (j in i + 1..4) 
            if (nomes[i].compareTo(nomes[j], ignoreCase = true) > 0) {
                x = nomes[i]
                nomes[i] = nomes[j]
                nomes[j] = x
            }

    println()
    for (i in 0..4)
        println("${i + 1}º nome: ${nomes[i]}")

    print("\nTecle <Enter> para encerrar... ")
    readLine()
}
